src/main/kotlin/binaryTree/PreorderTraverse.kt

```diff
@@ -0,0 +1,24 @@
+package binaryTree
+
+/**
+ * This is the preorder traverse of binary tree
+ * @param root - root of TreeNode
+ * @see TreeNode - node has reference on left and right child
+ */
+
+
+fun preorderTraverse(root: TreeNode?): List<Int> {
+    root ?: return listOf()
```

Comment:
You can use an immutable **emptyList()** directly.

It is more idiomatic and avoids unnecessary allocation.

```suggestion
    root ?: return emptyList()
```